### ISSUE TITLE
Help to active Titiler routes in the stac-fastapi new version

### ISSUE THREAD

**Issue by csaybar** [stac-utils/stac-fastapi#113]
First off, thanks so much for this superb Python library. I have a small dataset that I would like to make public available to everyone using stac-api.

I was trying to activate the OGC and Titiler routes in my STAC FastAPI app, like in this [video](https://www.youtube.com/watch?v=iP0oQZzupnQ).

```python
from stac_api.config import ApiSettings
from stac_api.api import create_app

settings = ApiSettings(
    add_ons=["tiles"]
)
app = create_app(settings)
```
But it seems that the API changes a bit, I was trying to solve it adding `TilesExtension` to the `stac_fastapi/server/app.py`:

```python
from stac_fastapi.api.app import StacApi
from stac_fastapi.extensions.core import (
    FieldsExtension,
    QueryExtension,
    SortExtension,
    TransactionExtension,
)
from stac_fastapi.extensions.third_party import TilesExtension, BaseTilesClient
from stac_fastapi.extensions.third_party import BulkTransactionExtension
from stac_fastapi.sqlalchemy.config import SqlalchemySettings
from stac_fastapi.sqlalchemy.core import CoreCrudClient
from stac_fastapi.sqlalchemy.session import Session
from stac_fastapi.sqlalchemy.transactions import (
    BulkTransactionsClient,
    TransactionsClient,
)

settings = SqlalchemySettings()
session = Session.create_from_settings(settings)
api = StacApi(
    settings=settings,
    extensions=[
        TilesExtension(client=BaseTilesClient()),
        TransactionExtension(client=TransactionsClient(session=session)),
        BulkTransactionExtension(client=BulkTransactionsClient(session=session)),
        FieldsExtension(),
        QueryExtension(),
        SortExtension()
    ],
    client=CoreCrudClient(session=session),
)
app = api.app
```

But it doesn't work. I'm a very basic user, sorry if it is a silly question, but I will over appreciated any help.

**Comment by geospatial-jeff** [comment]
@csaybar thanks for the issue!  It wouldn't surprise me if the switch to submodules broke the titiler integration, as there aren't any test cases for this feature.  I'll give it a try myself to see whats up!

**Comment by geospatial-jeff** [comment]
I see what is going on.  The `BaseTilesClient` that you are passing as an argument to the `TilesExtension` just defines the base pattern that may be implemented to create your own tiles extension.  In the previous version there was also an implementation of this (https://github.com/stac-utils/stac-fastapi/blob/1.0.0/stac_api/clients/tiles/ogc.py#L16-L41) that works with the sqlalchemy backend, but this code was not moved over into the submodule.

I'll have a PR soon to add this code back in, and maybe update our titiler version + add some test cases while I'm at it.

**Comment by geospatial-jeff** [comment]
When we switched to namespace packages the `TilesClient` (the actual implementation of `BaseTilesClient`) was dropped.  This is fixed in #118, once merged you should be able to enable the tiles extension as follows.

```python
from stac_fastapi.api.app import StacApi
from stac_fastapi.extensions.third_party.tiles import TilesClient, TilesExtension
from stac_fastapi.sqlalchemy.config import SqlalchemySettings
from stac_fastapi.sqlalchemy.core import CoreCrudClient
from stac_fastapi.sqlalchemy.session import Session

settings = SqlalchemySettings()
session = Session.create_from_settings(settings)
db_client = CoreCrudClient(session=session)
api = StacApi(
    settings=settings,
    extensions=[
        TilesExtension(TilesClient(db_client)),
    ],
    client=db_client,
)
app = api.app
```

Really the only change from your code is we are passing the `TilesClient` as a parameter to `TilesExtension`.  Because the tiles client also requires the database client, its worth pulling that out to global scope so you can re-use it.  Once this is set up, when I navigate to one of the `tiles` URLs I see the titiler viewer as expected:

![image](https://user-images.githubusercontent.com/40916268/112922781-b2aac500-90d2-11eb-8c7c-29b2f1706af7.png)

**Comment by csaybar** [comment]
Thank you @geospatial-jeff it works like a swarm!
